### PR TITLE
Check private key

### PR DESCRIPTION
**Pull request by ZainabF92**
Fixes: [eclipse-openj9/openj9#15413](https://github.com/eclipse-openj9/openj9/issues/15413)

Signed-off-by: Zainab Fatmi <zainab@ibm.com>

### CODE REVIEW

File: closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java [hunk 1]
```diff
@@ -129,6 +130,8 @@ protected void engineInit(Key key, SecureRandom random)
         } else {
             this.javaImplementation = null;
         }
+
+        ECUtil.checkPrivateKey(this.privateKey);
```
Comment (keithc-ca) [inline closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java:134]: Is there a reason this couldn't (or shouldn't) happen earlier, before line 107?

Comment (ZainabF92) [reply closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java]: moved the check to line 107

**Comment by keithc-ca** [comment]
jenkins test extended.openjdk xlinux jdk18

**Comment by pshipton** [comment]
The original version of this change was ported to jdk19+. Does it also need to be backported? The ECDSAPrvGreaterThanOrder.java test only exists in jdk18+, but the original feature was added to jdk8+ and may have the same problem.

**Comment by pshipton** [comment]
For the record it was introduce via https://github.com/ibmruntimes/openj9-openjdk-jdk18/commit/ca068f817e21fb

**Comment by pshipton** [comment]
jenkins compile amac jdk18

**Comment by pshipton** [comment]
extended.openjdk takes a while to run, and the machines aren't setup up correctly to pass all the testing. In particular there is jdk_nio Datagram testing which is running and timing out.

I've run a grinder on all the openjdk sanity/extended testing that looks security related.
`testList TESTLIST=jdk_security1,jdk_security2,jdk_security3,jdk_security4,jdk_security_infra,jdk11_tier1_cipher`
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/1130

ECKeyCheck.java is still failing but that is a different known problem https://github.com/eclipse-openj9/openj9/issues/15412
ECDSAPrvGreaterThanOrder.java is passing as expected with this fix.

**Pull request merged**